assets/js/loadpoint-settings.ts

```diff
@@ -0,0 +1,130 @@
+
+const getStoredSettings = (): LoadpointSettings => {
+  try {
+    const stored = localStorage.getItem(STORAGE_KEY);
+    if (stored) {
+      return JSON.parse(stored);
+    }
+  } catch (error) {
+    console.warn("Failed to parse stored loadpoint settings:", error);
+  }
+  return { order: [], visibility: {} };
+};
+
+const saveSettings = (settings: LoadpointSettings) => {
+  try {
+    localStorage.setItem(STORAGE_KEY, JSON.stringify(settings));
+  } catch (error) {
+    console.warn("Failed to save loadpoint settings:", error);
+  }
+};
+
+const initialSettings = getStoredSettings();
+
+const loadpointSettings = reactive({
+  order: initialSettings.order,
+  visibility: initialSettings.visibility,
+});
+
+// Watch for changes and persist to localStorage
+watch(
+  loadpointSettings,
+  (newSettings) => {
+    saveSettings({
+      order: newSettings.order,
+      visibility: newSettings.visibility,
+    });
+  },
+  { deep: true }
+);
+
+const initializeLoadpointSettings = (loadpointCount: number) => {
+  // Initialize order if not set or if loadpoint count changed
+  if (loadpointSettings.order.length === 0 || loadpointSettings.order.length !== loadpointCount) {
+    loadpointSettings.order = Array.from({ length: loadpointCount }, (_, i) => i);
+  }
+
+  // Initialize visibility if not set (default to visible)
+  for (let i = 0; i < loadpointCount; i++) {
+    if (!(i in loadpointSettings.visibility)) {
+      loadpointSettings.visibility[i] = true;
+    }
+  }
+};
+
+const getLoadpointOrder = () => loadpointSettings.order;
+
+const setLoadpointOrder = (order: number[]) => {
+  loadpointSettings.order = [...order];
+};
+
+const getLoadpointVisibility = (index: number): boolean => {
+  return loadpointSettings.visibility[index] ?? true;
+};
+
+const setLoadpointVisibility = (index: number, visible: boolean) => {
+  loadpointSettings.visibility[index] = visible;
+};
+
+const getOrderedVisibleLoadpoints = <T>(loadpoints: T[]): Array<T & { id: number }> => {
+  if (loadpoints.length === 0) return [];
+
+  initializeLoadpointSettings(loadpoints.length);
+
+  return loadpointSettings.order
+    .filter((index) => index < loadpoints.length && getLoadpointVisibility(index))
+    .map((index) => ({ ...loadpoints[index], id: index + 1 }))
+    .filter((item): item is T & { id: number } => item !== undefined);
+};
+
+const getLoadpointDisplayList = <T extends { title?: string }>(
+  loadpoints: T[]
+): Array<LoadpointDisplayItem & { originalData: T }> => {
+  if (loadpoints.length === 0) return [];
+
+  initializeLoadpointSettings(loadpoints.length);
+
+  return loadpointSettings.order
+    .filter((index) => index < loadpoints.length)
+    .map((index) => {
+      const loadpoint = loadpoints[index];
+      if (!loadpoint) {
+        throw new Error(`Loadpoint at index ${index} is undefined`);
+      }
+      return {
+        index,
+        visible: getLoadpointVisibility(index),
+        title: loadpoint.title || `Loadpoint ${index + 1}`,
```

Comment:
this has nothing to do with sorting. it duplicates existing fallback logic: https://github.com/evcc-io/evcc/blob/master/assets/js/components/Loadpoints/Loadpoint.vue#L242

Suggestion:
- on top level (site) lets convert the raw loadpoint (from api, currently strangely named LoadpointCompact) to a format ready for display (maybe DisplayLoadpoint) which as added `id` and resolved `title` (later consolidate other calculations like icon choice, ...).
- then just sort and filter `DisplayLoadpoint[]` > `DisplayLoadpoint[]`

We can pass the DisplayLoadpoint[] into the settings dialog.